### PR TITLE
[fix](memory) Fix LRUCacheType::NUMBER charge

### PR DESCRIPTION
**Pull request by xinyiZzz**
## Proposed changes

if LRUCacheType::NUMBER, charge not add handle_size, because charge at this time is no longer the memory size, but an independent weight.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

File: be/src/olap/lru_cache.cpp [hunk 1]
```diff
@@ -356,8 +356,12 @@ Cache::Handle* LRUCache::insert(const CacheKey& key, uint32_t hash, void* value,
     e->deleter = deleter;
     e->charge = charge;
     e->key_length = key.size();
-    e->total_size = (_type == LRUCacheType::SIZE ? handle_size + charge : 1);
+    // if LRUCacheType::NUMBER, charge not add handle_size,
+    // because charge at this time is no longer the memory size, but an weight.
+    e->total_size = (_type == LRUCacheType::SIZE ? handle_size + charge : charge);
```
Comment (yiguolei) [inline be/src/olap/lru_cache.cpp:361]: why  lRUTYPE::number need charge??

File: be/src/olap/lru_cache.cpp [hunk 2]
```diff
@@ -670,7 +674,7 @@ void ShardedLRUCache::update_cache_metrics() const {
 Cache::Handle* DummyLRUCache::insert(const CacheKey& key, void* value, size_t charge,
                                      void (*deleter)(const CacheKey& key, void* value),
                                      CachePriority priority, size_t bytes) {
-    size_t handle_size = sizeof(LRUHandle) - 1 + key.size();
+    size_t handle_size = sizeof(LRUHandle);
```
[the remark this answers is not in the document]
Comment (xinyiZzz) [reply be/src/olap/lru_cache.cpp:677]: core是因为我改成 new LRUHandle，release的时候是 free handle，所以这里必须用malloc

**Review by yiguolei — APPROVED** [review]

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**